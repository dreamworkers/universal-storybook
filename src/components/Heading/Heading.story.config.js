import PropTypes from "prop-types";

/**
 * Common story configuration
 */

export const ArgTypesHeading = {
  label: {
    control: {
      type: "text",
    },
  },
  color: { control: "color" },
  size: {
    options: ["small", "medium", "large"],
    control: {
      type: "radio",
    },
  },
};

export const ArgsHeading = {
  color: null,
  size: "medium",
};

export const ArgsHeadingLarge = {
  label: "Heading",
  ...ArgsHeading,
  size: "large",
};
export const ArgsHeadingMedium = {
  label: "Heading",
  ...ArgsHeading,
};
export const ArgsHeadingSmall = {
  label: "Heading",
  ...ArgsHeading,
  size: "small",
};

/**
 * React story configuration
 */

export const PropsHeading = {
  primary: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.oneOf(["small", "medium", "large"]),
  label: PropTypes.string.isRequired,
};
