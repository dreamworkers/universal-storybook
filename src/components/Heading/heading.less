@import "../../globals/tokens.less";

h1 {
  font-family: @font-family-base;
  font-weight: 700;
  line-height: 1.3;
}

.button {
  &--small {
    font-size: 14px;
  }
  &--medium {
    font-size: 18px;
  }
  &--large {
    font-size: 24px;
    font-weight: 900;
  }
}
