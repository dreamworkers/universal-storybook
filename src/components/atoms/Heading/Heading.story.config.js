import PropTypes from 'prop-types';

/**
 * Common story configuration
 */

export const ArgTypesHeading = {
  content: {
    control: {
      type: 'text',
    },
  },
  color: { control: 'color' },
  size: {
    options: ['small', 'medium', 'large'],
    control: {
      type: 'radio',
    },
  },
};

export const ArgsHeading = {
  color: null,
  size: 'medium',
};

export const ArgsHeadingLarge = {
  content: 'Heading',
  ...ArgsHeading,
  size: 'large',
};
export const ArgsHeadingMedium = {
  content: 'Heading',
  ...ArgsHeading,
};
export const ArgsHeadingSmall = {
  content: 'Heading',
  ...ArgsHeading,
  size: 'small',
};

/**
 * React story configuration
 */

export const PropsHeading = {
  primary: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  content: PropTypes.string.isRequired,
};
