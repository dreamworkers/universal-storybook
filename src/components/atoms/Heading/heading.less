@import '@Styles/inc/tokens.less';
@import '@Styles/inc/tokens-ext.less';

h1 {
  font-family: @family-font-secondary;
  font-weight: @weight-font-semi-bold;
  line-height: @line-height-font-base;
  letter-spacing: @letter-spacing-font-wide;
}

.heading {
  &--small {
    font-size: @size-font-medium;
  }
  &--medium {
    font-size: @size-font-large;
  }
  &--large {
    font-size: @size-font-extra-large;
    font-weight: @weight-font-bold;
  }
}
