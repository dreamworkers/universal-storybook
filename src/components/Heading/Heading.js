import React from "react";

import { PropsHeading, ArgsHeading } from "./Heading.story.config";
import "./heading.less";

/**
 * Primary UI component for user interaction
 */
export const Heading = ({ color, size, label, ...props }) => {
  return (
    <h1
      className={["button", `button--${size}`].join(" ")}
      style={color && { color }}
      {...props}
    >
      {label}
    </h1>
  );
};

Heading.propTypes = { ...PropsHeading };
Heading.defaultProps = { ...ArgsHeading };
