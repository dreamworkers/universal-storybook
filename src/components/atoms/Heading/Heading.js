import React from 'react';

import { PropsHeading, ArgsHeading } from './Heading.story.config';
import './heading.less';

/**
 * Primary UI component for user interaction
 */
export const Heading = ({ color, size, content, ...props }) => (
  <h1
    className={['heading', `heading--${size}`].join(' ')}
    style={color && { color }}
    {...props}
  >
    {content}
  </h1>
);

Heading.propTypes = { ...PropsHeading };
Heading.defaultProps = { ...ArgsHeading };
