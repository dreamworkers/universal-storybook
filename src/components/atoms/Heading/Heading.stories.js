import React from 'react';

import { Heading } from './Heading';
import {
  ArgTypesHeading,
  ArgsHeadingMedium,
  ArgsHeadingLarge,
  ArgsHeadingSmall,
} from './Heading.story.config';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  title: 'React/Atoms/Heading',
  component: Heading,
  argTypes: ArgTypesHeading,
};

const Template = (args) => <Heading {...args} />;

export const Large = Template.bind({});
Large.args = ArgsHeadingLarge;

export const Medium = Template.bind({});
Medium.args = ArgsHeadingMedium;

export const Small = Template.bind({});
Small.args = ArgsHeadingSmall;
