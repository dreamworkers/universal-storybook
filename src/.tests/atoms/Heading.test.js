import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { Heading } from "../../components/atoms/Heading/Heading";

configure({ adapter: new Adapter() });

describe("<Heading>", () => {
  describe("Heading will render", () => {
    it("creates a button component", () => {
      const props = {
        content: "Heading",
      };
      const tree = shallow(<Heading {...props}></Heading>);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Will take in multiple Props", () => {
    let tree;
    beforeAll(() => {
      const props = {
        color: null,
        size: "medium",
        content: "Heading",
      };
      tree = shallow(<Heading {...props}></Heading>);
    });

    it("Will have a child text", () => {
      expect(tree.text()).toEqual("Heading");
    });
  });
});
