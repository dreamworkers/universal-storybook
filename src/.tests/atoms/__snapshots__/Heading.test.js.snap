// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> Heading will render creates a button component 1`] = `ShallowWrapper {}`;
